linkerd2-proxy/src/rt.rs

```diff
@@ -1,11 +1,39 @@
 use tokio::runtime::{Builder, Runtime};
+use tracing::{debug, warn};
 
 #[cfg(feature = "multicore")]
 pub(crate) fn build() -> Runtime {
-    // The proxy creates an additional admin thread, but it would be wasteful to allocate a whole
-    // core to it; so we let the main runtime consume all cores the process has. The basic scheduler
-    // is used when the threaded scheduler would provide no benefit.
-    match num_cpus::get() {
+    // The proxy creates an additional admin thread, but it would be wasteful to
+    // allocate a whole core to it; so we let the main runtime consume all
+    // available cores. The number of available cores is determined by checking
+    // the environment or by inspecting the host or cgroups.
+    //
+    // The basic scheduler is used when the threaded scheduler would provide no
+    // benefit.
+    let mut cores = std::env::var("LINKERD2_PROXY_CORES")
+        .ok()
+        .and_then(|v| {
+            let opt = v.parse::<usize>().ok().filter(|n| *n > 0);
+            if opt.is_none() {
+                warn!(LINKERD2_PROXY_CORES = %v, "Ignoring invalid configuration");
+            }
+            opt
+        })
+        .unwrap_or(0);
+
+    let cpus = num_cpus::get();
+    debug_assert!(cpus > 0, "At least one CPU must be available");
+    if cores > cpus {
+        warn!(
+            cpus,
+            LINKERD2_PROXY_CORES = cores,
+            "Ignoring configuration due to insufficient resources"
+        );
+        cores = cpus;
+    }
+
+    debug!(cores, "Initializing runtime");
```

Comment:
TIOLI maybe this should be
```suggestion
    info!(cores, "Initializing runtime");
```
so that it gets logged in the default configuration? since it only gets logged once, there's no way to get the proxy to print it _after_ it's started up, so i feel like it may as well be there.